### ISSUE TITLE
Low timeout loader function race

### ISSUE THREAD

**Issue by jspri** [ReneKroon/ttlcache#44]
After running this for a while I noticed it's possible to get a key not found error (`ErrNotFound`), even when using a loader function. Repro below.

```go
// Cache sometimes returns key not found under parallel access with a loader function
func TestCache_TestLoaderFunctionParallelKeyAccess(t *testing.T) {
	t.Parallel()
	cache := NewCache()

	cache.SetLoaderFunction(func(key string) (data interface{}, ttl time.Duration, err error) {
		time.Sleep(time.Millisecond*300)
		return "1", 1*time.Nanosecond, nil
	})


	wg := sync.WaitGroup{}
	errCount := uint64(0)
	for i:=0; i<200; i++ {
		wg.Add(1)
		go func() {
			defer wg.Done()
			value, found := cache.Get("foo")
			if value != "1" || found != nil { // Use an atomic to avoid spamming logs
				atomic.AddUint64(&errCount, 1)
			}
		}()

	}

	wg.Wait()

	assert.Equalf(t, uint64(0), errCount, "expected 0 errs, got %d", errCount)

	cache.Close()
}
```

I spent a while trying to distill this down further, and I _think_ it occurs when:
- The TTL is low (10ms when I first noticed the bug - probably low enough that it could be triggered by a GC cycle)
- Multiple goroutines try to access the same key at the same time
- Some goroutines block while waiting for the loaderFunction
- By the time they unblock the freshly set key has already expired!

I think it would be more consistent if the blocked goroutines **always** got the output of the loaderfunction that they were blocked by. I would probably use [singleflight](https://pkg.go.dev/golang.org/x/sync/singleflight), but there might be a cleaner way to do it.

If you think it's a good idea then I'm happy to put together a PR.

**Comment by ReneKroon** [comment]
Hello, i understand the issue and see that it can be fixed. I'd like to know what your usecase is though. The 'average' usecase i have in mind is caching network requests which would have a TTL of seconds or maybe high milliseconds. This mean that the described issue is not really a thing. 

If you can describe what you are doing (probably some caching internal to the application) then it helps me to take better decisions for all users in the future.

Also reasoning about fast expiration: on the one hand i can defend the current behaviour by saying that if it takes `>expirationTime` to access the key that it would indeed be expired. On the other hand, if a key is retrieved and parked for further processing by GC/lock's later on i would consider it valid. Therefore the `singleflight` option looks good.

Please go ahead with the PR.

**Comment by jspri** [comment]
Thanks, I'll float a PR by you in a day or two.

I was mistaken in the first post, my use case actually had:
-  A TTL of 5 seconds
- Somewhere in the range of 100-1000 requests per second per key across 10ish keys.
- The loaderfunction takes 5ms to execute on average, but maybe sometimes much longer when there are network issues.
- We are caching some mutable database state in our application, and should/could use a queue or similar instead of a cache but this was easy to implement and good enough!
- We have many worker nodes, each that cache some frequently used state.
